### DISCUSSION TITLE
custom weather module: execute script mutiple times

### DISCUSSION THREAD

**Discussion by voidptrx** [Alexays/Waybar#2255] [Q&A]
Hi, I am using Waybar on sway. i have a custom weather module. What i want to do is when i click on this module it will show the weather information of another city. is it possible to do this?

For example, by default the module shows the weather for New York. When I click on the module, it will show the weather for London. If I click again, it will return to New York.

---

`weather.sh`
```bash
#!/usr/bin/env bash

text=$(curl -s "https://wttr.in/$1?format=1")

if [[ $? == 0 ]]
then
    text=$(echo "$text" | sed -E "s/\s+/ /g")
    tooltip=$(curl -s "https://wttr.in/$1?format=%l:+%C+%c+%t+%w+%m")
    if [[ $? == 0 ]]
    then
        tooltip=$(echo "$tooltip" | sed -E "s/\s+/ /g")
        echo "{\"text\":\"$text\", \"tooltip\":\"$tooltip\"}"
        exit
    fi
fi

echo "{\"text\":\"Service Unavailable\", \"tooltip\":\"Service Unavailable\"}
```

---

`waybar/config.jsonc`
```json
"custom/weather": {
        "format": "{}",
        "format-alt": "{source $HOME/.config/waybar/scripts/weather.sh 'London' 2> /dev/null}",
        "return-type": "json",
        "exec": "$HOME/.config/waybar/scripts/weather.sh 'New York' 2> /dev/null",
        "interval": 300,
        "tooltip": true
    },
```

**Comment by MajorMuff** [comment]
Even though this is an old thread I thought this would make a nice exercise so I made [this](https://github.com/MajorMuff/dump/blob/main/weather.sh).

You can use an onclick event for the module and have it run the script with the `--next` option. This switches to the next location and displays the weather for that location. Running it without this option will simply refresh the weather for the already selected location. It uses a temp file to store the currently selected location so it will reset to the default on reboot.

I'm not a bash pro by any means so there's probably a lot of room for improvement. It works as expected though!

**Reply by voidptrx** [reply]
Thank you for the answer. It works exactly as I want in terminal but not in waybar. It gives an error when I run it in waybar:

```
  custom/weather: * Line 1, Column 1
  Syntax error: value, object or array expected.
 ```

I'm not a bash guru either and I haven't quite figured out how to solve it.
